core/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -18,23 +18,20 @@ import scala.concurrent.{ExecutionContext, Future}
 
 @implicitNotFound(
   "Cannot convert from ${A} to an Entity, because no EntityEncoder[${F}, ${A}] instance could be found.")
-trait EntityEncoder[F[_], A] { self =>
-
-  /** Convert the type `A` to an [[EntityEncoder.Entity]] in the effect type `F` */
-  def toEntity(a: A): F[Entity[F]]
-
-  /** Headers that may be added to a [[Message]]
-    *
-    * Examples of such headers would be Content-Type.
-    * __NOTE:__ The Content-Length header will be generated from the resulting Entity and thus should not be added.
-    */
-  def headers: Headers
-
+/**
+  * @param headers Headers that may be added to a [[Message]]. Examples of such
+  * headers would be Content-Type. __NOTE:__ The Content-Length header will be
+  * generated from the resulting Entity and thus should not be added.
+  */
+final class EntityEncoder[F[_], A] private (
```

Comment:
This change is so we can derive an `Eq` instance to test lawfulness.  Now that I've done it, I think I can do it without this change.  I'll probably roll this back unless someone really likes it better this way.

Reply:
I rolled it back.